### DISCUSSION TITLE
Flux bootstrap GitLab keeps failing

### DISCUSSION THREAD

**Discussion by ashishonnet** [fluxcd/flux2#4767] [Q&A] [answered]
Hi,

I'm bootstrapping Flux with GitLab, but it keeps on failing with reconciler/lookup error.
It passes all pre-checks:

```
root@aslme-i0115:~/app-deploy# flux check --pre 
► checking prerequisites
✔ Kubernetes 1.28.9 >=1.26.0-0
✔ prerequisites checks passed
```
Logs show below error:

**flux logs** 
```
2024-05-07T09:21:51.677Z error GitRepository/flux-system.flux-system - Reconciler error failed to checkout and determine revision: unable to clone 'ssh://git@git.xxxxx.co.in/casestudy/ashish.kumar': dial tcp: lookup git.xxxxx.co.in on 10.96.0.10:53: no such host
```
**flux events**

```root@aslme-i0115:~/app-deploy# flux events
LAST SEEN          	TYPE   	REASON            	OBJECT                   	MESSAGE                                                                                                                                                                            
52s (x17 over 116m)	Warning	GitOperationFailed	GitRepository/flux-system	failed to checkout and determine revision: unable to clone 'ssh://git@git.xxxxx.co.in/casestudy/ashish.kumar': dial tcp: lookup git.xxxxx.co.in on 10.96.0.10:53: no such host	
```
I ran nslookup with git site i.e. git.xxxxx.co.in, and it can reach the IP address.
Is there anything at Flux which overrides DNS entries and fetch the DNS server from its own cache(may be)?
Because, if we see the error then flux is pointing to old configured DNS server.

I tried setting variables like `export FLUX_DNS_RESOLVER=x.x.x.x` and `export RESOLV_CONF=/etc/resolv.conf`, but of no use since Flux keeps using old DNS server. Is there anything that can be checked?

P.S. I can manually perform git clone using ssh on the server, by providing my own private key.

**Comment by stefanprodan** [comment, marked as answer]
The error comes from the cluster (source-controller pod) not from your local machine, CoreDNS is probably not configured correctly.

**Reply by ashishonnet** [reply]
okay, I did not explored CoreDNS yet. Let me check it. Thanks!!

**Reply by ashishonnet** [reply]
Added DNS entry in CoreDNS, which resolved this issue. Thanks!!
